megameklab/src/megameklab/ui/infantry/CISpecializationView.java

```diff
@@ -98,8 +94,13 @@ private void filterSpecializations() {
             public boolean include(Entry<? extends SpecializationModel, ? extends Integer> entry) {
                 SpecializationModel specModel = entry.getModel();
                 TechAdvancement techAdvancement = specModel.getTechAdvancement(entry.getIdentifier());
-                if ((1 << entry.getIdentifier() == Infantry.TAG_TROOPS)
-                        && TestInfantry.maxSecondaryWeapons(getInfantry()) < 2) {
+                final int bitFlag = 1 << entry.getIdentifier();
+                if ((bitFlag == Infantry.TAG_TROOPS)
+                        && (TestInfantry.maxSecondaryWeapons(getInfantry()) < 2)) {
+                    return false;
+                }
+                if (((bitFlag == Infantry.PARATROOPS) || (bitFlag == Infantry.MOUNTAIN_TROOPS))
```

Comment:
```suggestion
                } else if (((bitFlag == Infantry.PARATROOPS) || (bitFlag == Infantry.MOUNTAIN_TROOPS))
```

Reply:
If the previous condition is true it exits the method, so there's no need for an else here.

Reply:
There's no need, but it tends to prevent misreads in future @neoancient. Chained elses mean that there's zero way to misread the statements, while multiple if statements can cause flow misreads.